src/wazuh_modules/vulnerability_scanner/tests/policyManager/policyManager_test.cpp

```diff
@@ -0,0 +1,196 @@
+{
+    const auto& configJson {nlohmann::json::parse(R"({
+    "enable": "yes",
+    "index-status": "yes",
+    "feed-update-interval": "60m",
+    "offline-url": "file:///var/algo.tar.gz"
+})")};
+    EXPECT_THROW(m_policyManager->validateAndConfigureVulnerabilityDetection(configJson), std::runtime_error);
+}
+
+TEST_F(PolicyManagerTest, invalidFieldIndexer)
+{
+    const auto& configJson {nlohmann::json::parse(R"({
+    "enabled": "yes",
+    "hosts": ["http://10.2.20.2:9200", "https://10.2.20.42:9200"],
+    "username": "user",
+    "password": "pwd",
+    "api_key": "",
+    "ssl": {
+      "certificate_authorities": ["/var/ossec/"],
+      "certificate": "cert",
+      "key": ""
+    }
+})")};
+    EXPECT_THROW(m_policyManager->validateAndConfigureVulnerabilityDetection(configJson), std::runtime_error);
+}
+
+TEST_F(PolicyManagerTest, invalidConfiguration)
+{
+    const auto& configJson {nlohmann::json::parse(R"({
+  "vulnerability-detection": {
+    "enabled": "yes",
+    "index-status": "yes",
+    "feed-update-interval": "60m",
+    "offline-url": "file:///var/algo.tar.gz"
+  },
+  "indexer": {
+    "enabled": "no",
+    "hosts": ["http://10.2.20.2:9200", "https://10.2.20.42:9200"],
+    "username": "user",
+    "password": "pwd",
+    "api_key": "",
+    "ssl": {
+      "certificate_authorities": ["/var/ossec/"],
+      "certificate": "cert",
+      "key": ""
+    }
+  }
+})")};
+    m_policyManager->loadConfiguration(configJson);
+    EXPECT_THROW(m_policyManager->validateConfiguration(), std::runtime_error);
+}
+
+TEST_F(PolicyManagerTest, invalidFieldIndexerSecondScenario)
+{
+    const auto& configJson {nlohmann::json::parse(R"({
+  "vulnerability-detection": {
+    "enabled": "yes",
+    "index-status": "yes",
+    "feed-update-interval": "60m",
+    "offline-url": "file:///var/algo.tar.gz"
+  },
+  "indexer": {
+    "enable": "yes",
+    "hosts": ["http://10.2.20.2:9200", "https://10.2.20.42:9200"],
+    "username": "user",
+    "password": "pwd",
+    "api_key": "",
+    "ssl": {
+      "certificate_authorities": ["/var/ossec/"],
+      "certificate": "cert",
+      "key": ""
+    }
+  }
+})")};
+    m_policyManager->loadConfiguration(configJson);
+    EXPECT_THROW(m_policyManager->validateConfiguration(), std::runtime_error);
+}
+
+TEST_F(PolicyManagerTest, validConfiguration)
+{
+    const auto& configJson {nlohmann::json::parse(R"({
+  "vulnerability-detection": {
+    "enabled": "yes",
+    "index-status": "yes",
+    "feed-update-interval": "60m",
+    "offline-url": "file:///var/algo.tar.gz"
+  },
+  "indexer": {
+    "enabled": "yes",
+    "hosts": ["http://10.2.20.2:9200", "https://10.2.20.42:9200"],
+    "username": "user",
+    "password": "pwd",
+    "api_key": "",
+    "ssl": {
+      "certificate_authorities": ["/var/ossec/"],
+      "certificate": "cert",
+      "key": ""
+    }
+  }
+})")};
+    m_policyManager->loadConfiguration(configJson);
+    EXPECT_NO_THROW(m_policyManager->validateConfiguration());
```

Comment:
We should validate the values parsed for each json key using the getters functions.

Reply:
We merge `validConfiguration` and `checkValidParameters`